dotnet/samples/Concepts/Agents/OpenAIAssistant_Streaming.cs

```diff
@@ -0,0 +1,144 @@
+/// Demonstrate creation of <see cref="ChatCompletionAgent"/> and
+/// eliciting its response to three explicit user messages.
+/// </summary>
+public class OpenAIAssistant_Streaming(ITestOutputHelper output) : BaseAgentsTest(output)
+{
+    private const string ParrotName = "Parrot";
+    private const string ParrotInstructions = "Repeat the user message in the voice of a pirate and then end with a parrot sound.";
+
+    [Fact]
+    public async Task UseStreamingChatCompletionAgentAsync()
+    {
+        // Define the agent
+        OpenAIAssistantAgent agent =
+            await OpenAIAssistantAgent.CreateAsync(
+                kernel: new(),
+                clientProvider: this.GetClientProvider(),
+                new(this.Model)
+                {
+                    Instructions = ParrotInstructions,
+                    Name = ParrotName,
+                    Metadata = AssistantSampleMetadata,
+                });
+
+        // Create a thread for the agent conversation.
+        string threadId = await agent.CreateThreadAsync(new OpenAIThreadCreationOptions { Metadata = AssistantSampleMetadata });
+
+        // Respond to user input
+        await InvokeAgentAsync(agent, threadId, "Fortune favors the bold.");
+        await InvokeAgentAsync(agent, threadId, "I came, I saw, I conquered.");
+        await InvokeAgentAsync(agent, threadId, "Practice makes perfect.");
+
+        // Output the entire chat history
+        await DisplayChatHistoryAsync(agent, threadId);
+    }
+
+    [Fact]
+    public async Task UseStreamingChatCompletionAgentWithPluginAsync()
+    {
+        const string MenuInstructions = "Answer questions about the menu.";
+
+        // Define the agent
+        OpenAIAssistantAgent agent =
+            await OpenAIAssistantAgent.CreateAsync(
+                kernel: new(),
+                clientProvider: this.GetClientProvider(),
+                new(this.Model)
+                {
+                    Instructions = MenuInstructions,
+                    Name = "Host",
+                    Metadata = AssistantSampleMetadata,
+                });
+
+        // Initialize plugin and add to the agent's Kernel (same as direct Kernel usage).
+        KernelPlugin plugin = KernelPluginFactory.CreateFromType<MenuPlugin>();
+        agent.Kernel.Plugins.Add(plugin);
+
+        // Create a thread for the agent conversation.
+        string threadId = await agent.CreateThreadAsync(new OpenAIThreadCreationOptions { Metadata = AssistantSampleMetadata });
+
+        // Respond to user input
+        await InvokeAgentAsync(agent, threadId, "What is the special soup?");
+        await InvokeAgentAsync(agent, threadId, "What is the special drink?");
+
+        // Output the entire chat history
+        await DisplayChatHistoryAsync(agent, threadId);
+    }
+
+    // Local function to invoke agent and display the conversation messages.
+    private async Task InvokeAgentAsync(OpenAIAssistantAgent agent, string threadId, string input)
```

Comment:
Same code as sample above. Should this be extracted, or maybe it's more readable for it to be duplicated in these concepts?

Reply:
This helper method (now) only exists in `ChatCompletion_Streaming` and  `OpenAIAssistant_Streaming`.  Both versions are slightly specific, in that the `OpenAIAssistant_Streaming` is managing a threadid and `ChatCompletion_Streaming` only manages `ChatHistory`.

The reason this exists as a helper is that it is called repeatedly from two different flavors of the sample (for each class):  one that uses plug-ins and one that does not.

I know that generally its been encouraged to keep the samples relatively "intact" for each of copying; although, I did get "permission" to introduce a `BaseAgentsTest` which does help with consistency and lesses duplication.